File: app/src/test/java/com/huynd/skyobserver/presenters/PriceOneDayPresenterTest.java
package com.huynd.skyobserver.presenters;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.fail;

/**
 * Created by HuyND on 8/16/2017.
 */

public class PriceOneDayPresenterTest {
    private PriceOneDayPresenter mPresenter;

    @Before
    public void setUp() throws Exception {
        mPresenter = new PriceOneDayPresenterImpl(null, null);
    }

    @Test
    public void onBtnGetPricesClickShouldDoNothingIfViewIsNull() throws Exception {
        try {
            mPresenter.getPrices(2017, 10, 01, "SGN", "HAN", true);
        } catch (Exception e) {
            fail("Unexpected behavior happened.");
        }
    }

    @Test
    public void onResponseShouldDoNothingIfViewIsNull() throws Exception {
        try {
            ((PriceOneDayPresenterImpl) mPresenter).onGetPricesResponse(null, true);
        } catch (Exception e) {
            fail("Unexpected behavior happened.");
        }
    }
}

File: app/src/test/java/com/huynd/skyobserver/presenters/PriceOneDayPresenterTest.kt
package com.huynd.skyobserver.presenters

import org.junit.Before
import org.junit.Test
import org.junit.Assert.fail

/**
 * Created by HuyND on 5/25/2019.
 */

class PriceOneDayPresenterTest {
    private lateinit var mPresenter: PriceOneDayPresenter

    @Before
    @Throws(Exception::class)
    fun setUp() {
        mPresenter = PriceOneDayPresenterImpl(null, null)
    }

    @Test
    fun onBtnGetPricesClickShouldDoNothingIfViewIsNull() {
        try {
            mPresenter.getPrices(2017, 10, 0, "SGN", "HAN", true)
        } catch (e: Exception) {
            fail("Unexpected behavior happened.")
        }
    }

    @Test
    fun onResponseShouldDoNothingIfViewIsNull() {
        try {
            (mPresenter as PriceOneDayPresenterImpl).onGetPricesResponse(null, true)
        } catch (e: Exception) {
            fail("Unexpected behavior happened.")
        }
    }
}
